### PR TITLE
NixOS compatibility

### PR DESCRIPTION
**Pull request by globin**
This fixes some minor issues in regard to NixOS compatibility mostly
due to the fact that NixOS only has the binaries `/usr/bin/env` and `/bin/sh`
in deterministic absolute locations.

Additionally this removes the copying of the template to the lxc template
path, which is in a read-only mount on NixOS and a non-deterministic location.
Instead it uses the absolute path to the template which is supported since lxc 1.0.

[Review on Reviewable](https://reviewable.io/reviews/fgrehm/vagrant-lxc/376)

### CODE REVIEW

**Comment by globin** [comment]
Fixed the failing test

**Comment by fgrehm** [comment]
Sorry for the long silence but as per GH-375 I'm looking for help to maintain this plugin and for now I'm holding back from bringing in new changes =/

If you are interested on maintaining the project and keep things moving please shoot me an email or drop a message on GH-375 so I can arrange things.

Cheers :beers:

**Comment by globin** [comment]
I've actually just started writing a comment on there, since we use this heavily @mayflower..

So we are definitely interested in keeping this alive and up to date.

I guess @fpletz, @codec, @fadenb and possibly @marco-jantke, might want to chime in, too.

What's the best way to support you?

**Comment by fgrehm** [comment]
That is **awesome**! :boom: :raised_hands: :muscle:  :beers: :fireworks: 

Well, if you guys are up for it I can provide write access to one of you and you can start hacking away, merging PRs, closing issues... :smile: I'll be around to help out and provide feedback whenever needed but unfortunately I'm very unlikely to have the time to test changes or write code. 

Once you guys are ready to cut a release, just open up PR and `@ping` me on it so that I can provide RubyGems.org access to someone over there too.

WDYT?

**Comment by fpletz** [comment]
@globin IMHO we should remove at least the `gemset.nix` because the sha256sums of some git repositories are not stable and there are still some problems with SSL. We should probably wait until this is fixed in NixOS and just get the code & test fixes merged ASAP.

File: lib/vagrant-lxc/provider.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def sudo_wrapper
 
       def ensure_lxc_installed!
         begin
-          sudo_wrapper.run("/usr/bin/which", "lxc-create")
+          sudo_wrapper.run("/usr/bin/env", "which", "lxc-create")
```
Comment (fpletz) [inline lib/vagrant-lxc/provider.rb]: Can we prepend `"/usr/bin/env"` to every path in the sudo_wrapper instead to fix this everywhere? There are some more similar cases in the networking code where tools like `ip` and `brctl` are called.

**Comment by globin** [comment]
Updated and rebased

**Comment by krtek4** [comment]
Definitively would love to see that in a release :)

**Comment by fpletz** [comment]
:+1:

**Comment by mvikharev** [comment]
That breaks avoiding sudo passwords on Ubuntu 14.04 when running vagrant ssh. 
```
$ cat /etc/sudoers.d/vagrant-lxc
# Automatically created by vagrant-lxc
mvikharev ALL=(root) NOPASSWD: /usr/local/bin/vagrant-lxc-wrapper

$ vagrant ssh
...
INFO machine: Initializing machine: default
INFO machine:   - Provider: Vagrant::LXC::Provider
INFO subprocess: Starting process: ["/usr/bin/sudo", "/usr/bin/env", "/usr/local/bin/vagrant-lxc-wrapper", "lxc-ls"]
INFO subprocess: Command not in installer, restoring original environment...
[sudo] password for mvikharev:
```

**Comment by ccope** [comment]
Also, I think the sudo wrapper is actually not compatible with nix. If it allows calling commands with /usr/bin/env, then an attacker could just add whatever binaries they want to the user's path and call them.

**Comment by globin** [comment]
Fix released in 1.2.1

**Pull request merged**